text/0000-near-shell.md

```diff
@@ -0,0 +1,449 @@
+
+`near-shell` is an excellent tool to attract and engage new NEAR developers, both core and contract level, and if it contains all the required features to select and configure and deploy NEAR networks, nodes, crypto assets (i.e., keys), accounts, and wallets, it can be the ideal single-entry point for those potential new hackers.  
+
+# Motivation
+[motivation]: #motivation
+
+These enhancements are intended to simplify and enhance developer and user productivity by providing a single command-line tool which exercises all major features of NEAR. As each major piece of NEAR functionality is added to `near-shell`, we should see utilization of each of those features increase.  Additionally, community mindshare and understanding of NEAR should increase as `near-shell` will contain detailed help documentation for each major feature with clear examples of use, just like Linux man pages and/or other major CLI tools which have integrated help for each command.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This NEP will jump straight into the concepts necessary to understand the components needed in `near-shell`, and generally an extensible CLI tool. Unlike other NEPs, the guide-level and reference-level explanations are combined. There is nothing overly technical in the following sections, but more conceptual in nature. With that, the discussion begins by separating the three types of stored information needed for a successful, long-term `near-shell`.
+
+## Settings, config, and key management
+
+Shell-experience **settings**, connection **configuration**, and **key management** are the three types of stored data necessary for developers to reliably build and deploy projects.
+ 
+```
+─ awesome-near-project  ⟵ NEAR dApp
+ ├── .near-config       ⟵ Stores shell-experience settings and connection configuration
+ │  ├── connections     ⟵ Contains files used to connect and deploy a contract (except keys)
+ │  │  ├── default.js   ⟵ Default configuration is for development
+ │  │  └── localnet.js  ⟵ Custom connection added by user for localnet
+ │  └── settings.js     ⟵ Stores near-shell settings, how shell behaves when run in this project
+ └── .near-credentials  ⟵ Previously "neardev" this directory contains private key inforamtion
```

Comment:
credentials for sure should not be here. this is both easiest way to delete them (delete & reclone the project) or expose them because forgot to add the folder into .gitignore.